### PR TITLE
Fix: Various YAML file path scenario

### PR DESCRIPTION
**Pull request by olblak**
Similar to https://github.com/updatecli/updatecli/pull/845
The current PR fixes the different file path scenario for the YAML resource

## Test

To test this pull request, you can run the following commands:

```shell
cp <to_package_directory>
go test
```

### CODE REVIEW

File: pkg/plugins/resources/yaml/source.go [hunk 1]
```diff
@@ -14,10 +13,9 @@ import (
 // Source return the latest version
 func (y *Yaml) Source(workingDir string) (string, error) {
 	// By default workingDir is set to local directory
-
-	if len(workingDir) > 0 {
-		y.spec.File = filepath.Join(workingDir, y.spec.File)
-	}
+	// Merge Filepath with current working dire, unless file is an http url
+	// To merge File path with current working dire, unless file is an http url
```
Comment (hervelemeur) [inline pkg/plugins/resources/yaml/source.go:17]:
```suggestion
```

File: pkg/plugins/resources/yaml/target.go [hunk 2]
```diff
@@ -22,16 +21,22 @@ func (y *Yaml) Target(source string, dryRun bool) (bool, error) {
...
 	return y.target(source, dryRun)
 }
 
 func (y *Yaml) target(source string, dryRun bool) (bool, []string, string, error) {
 	var files []string
 	var message string
 
+	if strings.HasPrefix(y.spec.File, "https://") ||
+		strings.HasPrefix(y.spec.File, "http://") {
+		return false, files, message, fmt.Errorf("URL scheme is not supported for XML target: %q", y.spec.File)
```
Comment (hervelemeur) [inline pkg/plugins/resources/yaml/target.go:37]:
```suggestion
		return false, files, message, fmt.Errorf("URL scheme is not supported for YAML target: %q", y.spec.File)
```

**Review by hervelemeur — CHANGES_REQUESTED** [review]

File: pkg/plugins/resources/yaml/utils.go [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package yaml
+
+import (
+	"path/filepath"
+	"strings"
+)
+
+// joinPathwithworkingDirectoryPath To merge File path with current working dire, unless file is an http url
+func joinPathWithWorkingDirectoryPath(fileName, workingDir string) string {
+	if workingDir != "" &&
```
[the remark this answers is not in the document]
Comment (olblak) [reply pkg/plugins/resources/yaml/utils.go:10]: That's a good suggestion, I'll implement your suggestion

**Comment by olblak** [comment]
So I messed up the git history with old xml change, I first need to clean this up

File: pkg/plugins/resources/yaml/condition.go [hunk 4]
```diff
@@ -18,9 +17,11 @@ func (y *Yaml) Condition(source string) (bool, error) {
 
 // ConditionFromSCM checks if a key exists in a yaml file
 func (y *Yaml) ConditionFromSCM(source string, scm scm.ScmHandler) (bool, error) {
-	if !filepath.IsAbs(y.spec.File) {
-		y.spec.File = filepath.Join(scm.GetDirectory(), y.spec.File)
+
```
Comment (lemeurherve) [inline pkg/plugins/resources/yaml/condition.go:20]:
```suggestion
```

File: pkg/plugins/resources/yaml/source.go [hunk 5]
```diff
@@ -14,10 +13,8 @@ import (
 // Source return the latest version
 func (y *Yaml) Source(workingDir string) (string, error) {
 	// By default workingDir is set to local directory
-
-	if len(workingDir) > 0 {
-		y.spec.File = filepath.Join(workingDir, y.spec.File)
-	}
+	// Merge Filepath with current working dire, unless file is an http url
```
Comment (lemeurherve) [inline pkg/plugins/resources/yaml/source.go:16]:
```suggestion
	// Merge File path with current workingDir, unless File is an HTTP URL
```

File: pkg/plugins/resources/yaml/utils.go [hunk 6]
```diff
@@ -0,0 +1,18 @@
+package yaml
+
+import (
+	"path/filepath"
+	"strings"
+)
+
+// joinPathwithworkingDirectoryPath To merge File path with current working dire, unless file is an http url
```
Comment (lemeurherve) [inline pkg/plugins/resources/yaml/utils.go:8]:
```suggestion
// joinPathwithworkingDirectoryPath To merge File path with current workingDir, unless file is an HTTP URL
```

File: pkg/plugins/resources/yaml/utils.go [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+func joinPathWithWorkingDirectoryPath(fileName, workingDir string) string {
+	if workingDir == "" ||
+		filepath.IsAbs(fileName) ||
+		strings.HasPrefix(fileName, "https://") ||
+		strings.HasPrefix(fileName, "http://") {
+		return fileName
+	}
+
+	return filepath.Join(workingDir, fileName)
```
Comment (lemeurherve) [inline pkg/plugins/resources/yaml/utils.go:17]:
```suggestion
func joinPathWithWorkingDirectoryPath(filePath, workingDir string) string {
	if workingDir == "" ||
		filepath.IsAbs(filePath) ||
		strings.HasPrefix(filePath, "https://") ||
		strings.HasPrefix(filePath, "http://") {
		return filePath
	}

	return filepath.Join(workingDir, filePath)
```

File: pkg/plugins/resources/yaml/utils_test.go [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+			workingDir:     "/opt",
+			expectedResult: "/opt/test.yaml",
+		},
+	}
+
+	for _, tt := range testData {
+		t.Run(tt.name, func(t *testing.T) {
+
+			gotResult := joinPathWithWorkingDirectoryPath(tt.fileName, tt.workingDir)
+
```
Comment (lemeurherve) [inline pkg/plugins/resources/yaml/utils_test.go:59]:
```suggestion
			gotResult := joinPathWithWorkingDirectoryPath(tt.fileName, tt.workingDir)
```
(nit)

**Comment by olblak** [comment]
The e2e tests won't work until this PR is merged

**Comment by hervelemeur** [comment]
FYI you can commit suggestions in batch ;)

**Comment by lemeurherve** [comment]
> The e2e tests won't work until this PR is merged

Did it worked on your machine? Here and on mine it didn't: https://github.com/updatecli/updatecli/runs/8284237135?check_suite_focus=true

**Pull request merged**